pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -128,6 +130,7 @@ var (
 		"Specify the RSA key size to use for self-signed Istio CA certificates.")
 
 	// TODO: Likely to be removed and added to mesh config
+	// deprecated - it is only used to enable k8s signing - use presence of K8S_SIGNER instead.
 	externalCaType = env.Register("EXTERNAL_CA", "",
 		"External CA Integration Type. Permitted value is ISTIOD_RA_KUBERNETES_API.").Get()
```

Comment:
If you don't have CERT_SIGNER_DOMAIN and K8S_SIGNER = there is nothing we can do, EXTERNA_CA will not work. 

If either is set - it is clear intent to use a K8S signer and there is no need for EXTERNAL_CA, just redundant setting.

Allowing an arbitrary cert-signer in the CSR request - without having a match to K8S_SIGNER or CERT_SIGNER_DOMAIN prefix at least ( even that is risky ) - huge security problem. You can't let clients use arbitrary signers in K8S. 

CERT_SIGNER_DOMAIN only works (maybe) with carefully written signers - for example CertManager will be pretty bad fit. Client can put arbitrary value - and would be able to sign with Issuers from any namespace they want.

Reply:
To be honest - I would deprecate CERT_SIGNER_DOMAIN and the hidden metadata that allows client to inject arbitrary signers as well, until there is a design document that is approved and reviewed for security ( including by CertManager and other signer providers). It got in as 'experimental, no need to look or review because velocity is great'